src/Microsoft.AspNet.OData.Shared/Routing/Conventions/EntitySetRoutingConvention.cs

```diff
@@ -42,6 +42,13 @@ internal static string SelectActionImpl(ODataPath odataPath, IWebApiControllerCo
                         "Post" + entitySet.EntityType().Name,
                         "Post");
                 }
+                else if (ODataRequestMethod.Patch == controllerContext.Request.Method)
+                {
+                    // e.g. Try PatchCustomer first, then fall back to Patch action name
+                    return actionMap.FindMatchingAction(
+                        "Patch" + entitySet.EntityType().Name,
```

Comment:
>"Patch" + entitySet.EntityType().Name [](start = 24, length = 37)

I think it should be "PatchCustomers" (the entityset name), not "PatchCustomer" (the entitytype name).

Reply:
Besides, how about the cast?

---
In reply to: [580534068](https://github.com/OData/WebApi/pull/2275#discussion_r580534068) [](ancestors = 580534068)

Reply:
+1 on the entity set name -- good catch!